src/journal/journald-server.c

```diff
@@ -1156,12 +1193,77 @@ void server_dispatch_message(
         dispatch_message_real(s, iovec, n, m, c, tv, priority, object_pid);
 }
 
+static int do_flush(Server *s, sd_journal *j) {
+        usec_t before_flush;
+        unsigned n = 0;
+        int r;
+
+        assert(s && s->system_journal);
+        assert(j);
+
+        before_flush = now(CLOCK_MONOTONIC);
+
+        SD_JOURNAL_FOREACH(j) {
+                JournalFile *t = NULL, *f = j->current_file;
+                Object *o = NULL;
+                uid_t uid = 0;
+
+                assert(f && f->current_offset > 0);
+
+                r = journal_file_move_to_object(f, OBJECT_ENTRY, f->current_offset, &o);
+                if (r < 0)
+                        return log_error_errno(r, "Can't read entry: %m");
+
+                r = journal_file_parse_uid_from_filename(f->path, &uid);
```

Comment:
does this actually make sense? wouldn't it make more sense to use the _UID= field of the record instead?

Reply:
Hmm.. it depends if we want to keep the current behavior or not I would say. Currently all entries in the volatile (system) journal are flushed into /var/log/journal/xxx/system.journal, regardless of its _UID=.

With this implementation we would keep the splitting when flushing only if it was enabled for volatile.